SchedulerJobs/SchedulerJobs.Services/ELinksService.cs

```diff
@@ -88,16 +93,17 @@ public async Task ImportLeaversJudiciaryPeopleAsync(DateTime fromDate)
                     _logger.LogWarning($"ImportJudiciaryLeavers: No of leavers who are invalid '{invalidCount}' in page '{currentPage}'.");
                     _logger.LogInformation($"ImportJudiciaryLeavers: Calling bookings API with '{leaversResult.Count}' leavers");
 
-                    var response = await _bookingsApiClient.BulkJudiciaryLeaversAsync(leaversResult.Select(x =>  JudiciaryLeaverRequestMapper.MapTo(x)));
+                    var response = await _bookingsApiClient.BulkJudiciaryLeaversAsync(leaversResult.Select(x => JudiciaryLeaverRequestMapper.MapTo(x)));
                     response?.ErroredRequests.ForEach(x => _logger.LogError("ImportJudiciaryLeavers: {ErrorResponseMessage}", x.Message));
 
                     currentPage++;
                 }
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "There was a problem importing judiciary leavers");
-                throw;
+                catch (Exception ex)
+                {
+                    currentPage++;
```

Comment:
Just double checking if this is intentional, so we are saying that if an exception is being thrown, we should still continue to be in an infinite loop here until leaversResult's count is zero explicitly? And same as before, do we have unit test to ensure leaversResult being null wont cause infinite loop here?